### PR TITLE
add helpful error message for when file path doesnt exist and give hint for sysglob for multiple files

### PR DESCRIPTION
**Pull request by taylorreiter**
Addresses #49. Came up when @ecpierce was running through the vignette with her own samples.

### CODE REVIEW

File: tests/testthat/test-read_sourmash_outputs.R
Comment (mertcelebi) [inline tests/testthat/test-read_sourmash_outputs.R:4]: love tests

File: R/read_sourmash_outputs.R
Comment (mertcelebi) [inline R/read_sourmash_outputs.R:8]: probably not worth it now, but may be worth separating this into a `utils/` directory or smth

File: R/read_sourmash_outputs.R
Comment (mertcelebi) [inline R/read_sourmash_outputs.R:218]: I wish there was a way to abstract these read methods so we didn't have to repeat all of this, but it's fine for now i guess

File: R/read_sourmash_outputs.R
Comment (mertcelebi) [inline R/read_sourmash_outputs.R:10]: It's generally not great to make these checks in a nested way, because it can get tough to follow for the reader (let's say you add a check for S3 URIs in the future, it'd have to be nested one-level deeper). 

The heuristics I use for this type of stuff is: (a) it's better to return early from validation methods and (b) following that, checking for something is valid (rather than not valid) leads to more linear checks, rather than more nested checks

With these heuristics, here's how I would write this method (it's more verbose, but clearer imo):

```
check_file_and_url_one <- function(file){
  # check if file or URL are valid, and if not, print helpful messages and errors.
  if (file.exists(file)) {
    return(NULL)
  }

  message("File does not exist locally. Checking as url. If you did not pass a url, please provide a valid file path and retry. The Sys.glob() function might help if you're specifying multiple file paths.")
  if(httr::HEAD(file)$status == 200) {
    return(NULL)
  }

  stop("Neither file path nor url are valid.")
}
```

**Review by mertcelebi — APPROVED** [review]
Approving with a couple small comments

File: R/read_sourmash_outputs.R
Comment (taylorreiter) [reply R/read_sourmash_outputs.R:10]: ooooh i love this. I didn't think of doing a `return(NULL)`, and that solves a lot of problems I was having with the logic of this. I'll make this change before merging!

File: R/read_sourmash_outputs.R
Comment (taylorreiter) [reply R/read_sourmash_outputs.R:6]: ok I'll keep this in mind for the future and move toward that!

File: R/read_sourmash_outputs.R
Comment (austinhpatton) [reply R/read_sourmash_outputs.R:220]: I'll second that - even minimizing repetition would help to improve legibility at first pass, but probably not an essential change.

**Review by austinhpatton — APPROVED** [review]
No major comments on top of what Mert said - looks good!

**Pull request merged**
